### DISCUSSION TITLE
Trimmed leading zeros formatting timestamp milliseconds

### DISCUSSION THREAD

**Discussion by k4hvd1** [dbeaver/dbeaver#12772] [Q&A] [answered]
If dataformat for timestamp (Prefferences/DataEditor/Dataformat) shows milliseconds ( **S** specifier)  e.g. yyyy-MM-dd HH:mm:ss.**S**  than following leading zeroes are wrongly trimmed. Therefore, following value will be shown wrongly:

value_:   2021-04-14 09:17:50.057 
shown:  2021-04-14 09:17:50.57

Workaround is to use **mmm** or **f** for seconds fractions (for miliseconds) .

The docs https://dbeaver.com/docs/wiki/Managing-Data-Formats/ should be updated as the format **mmm** is not mentioned but it works.

#### System information: 
- Windows Server 2016 Datacenter (AWS workspace)
- DBeaver version: 21.0.0

#### Connection specification:
- Snowflake
- No tunnels...

#### Describe the problem you're observing:
Date is not properly presented for timestamp (snowflake database) 2021-04-14 09:17:50.**057** -> 2021-04-14 09:17:50.**57**
#### Steps to reproduce, if exist:
Go to  -> Prefferences/DataEditor/Dataformat  
Set timestamp format :  yyyy-MM-dd HH:mm:ss.S  
Create an DataTIme data that contains second fraction e.g. 2021-04-14 09:17:50.**057** 
View this data it will show 2021-04-14 09:17:50.**57**
Or even View Table , and try to enter second fraction **.057** it will automatically change to **.57**

**Comment by kseniiaguzeeva** [comment, marked as answer]
Sorry for the long response. I cannot reproduce it. Looks like a syntax mistake. Please try this template
`yyyy-MM-dd HH:mm:ss.SSS`
![Screenshot 2021-05-13 at 11 56 09](https://user-images.githubusercontent.com/51405061/118103509-aa88b980-b3e2-11eb-855c-3dd3c78a829e.png)

**Comment by k4hvd1** [comment]
Please, follow the steps that were described.

**Comment by uslss** [comment]
it is needed to use **SSS** to show leading zeros.  So if you use **S** leading zeros are omitted to reduce the value to the desired width.
https://bugs.java.com/bugdatabase/view_bug.do?bug_id=4516915
